### DISCUSSION TITLE
Inconsistent boolean behaviour

### DISCUSSION THREAD

**Discussion by rihardsgrislis** [laravel/framework#33692] [General]
Hey,

Is there any specific reason why `boolean` validation rule doesn't accept `"true"`, `"on"`, `"yes"` but boolean helper method does?

**Validation:** 
https://laravel.com/docs/7.x/validation#rule-boolean

> **boolean**
> The field under validation must be able to be cast as a boolean. Accepted input are true, false, 1, 0, "1", and "0".

**Requests:**
https://laravel.com/docs/7.x/requests#retrieving-input

> For convenience, you may use the boolean method to retrieve these values as booleans. The boolean method returns true for 1, "1", true, "true", "on", and "yes".

**Issue I ran into:**
A POST request is made to my service using form-data payload and `true`/`false` values are turned into `"true"`/`"false"`. Using the boolean helper I wouldn't care wheter the request payload was sent as form data or JSON. But I can't set the validation rule to "boolean" because it won't accept `"true"`/`"false"`.

Wouldn't it be great if boolean validation rule accepted "true", "on", and "yes" like the helper does thus aligning the behaviour of both?

**P.S.** For the boolean helper method Laravel uses PHPs `filter_var` with `FILTER_VALIDATE_BOOLEAN`.

**Comment by ahinkle** [comment]
Feel free to send a PR - this could be a breaking change as old valid values could potentially be considered invalid visa/versa - I would suggest sending it to the master branch.

The behavior of `$request->boolean()` [was added earlier this year](https://github.com/laravel/framework/pull/31160) to use `FILTER_VALIDATE_BOOLEAN`.
